src/Mailtrap.php

```diff
@@ -110,15 +117,28 @@ public function receiveAnEmail($params)
      *
      * @return array
      */
-    public function fetchLastMessage()
+    public function searchForMessage($emailSearchForString, $inboxID)
     {
-        $messages = $this->client->get("inboxes/{$this->config['inbox_id']}/messages")->getBody();
-        if ($messages instanceof Stream) {
-            $messages = $messages->getContents();
-        }
-        
-        $messages = json_decode($messages, true);
+        $counter = 0;
+        if ($inboxID != ''){ // test sent a specific email box to search
+            do {
+                sleep(1);
+                $counter++;
+                echo " Counter = " . $counter . " : ";              
+                $messages = $this->client->get("inboxes/$inboxID/messages?search=".$emailSearchForString)->getBody();
+                $messages = json_decode($messages, true);
+            } while ($counter < 60 && $messages == Null);
 
+           
+       } else { // Use the config email box
+            do {
+                sleep(1);
+                $counter++;
+                echo " Counter = " . $counter;
+                $messages = $this->client->get("inboxes/{$this->config['inbox_id']}/messages?search=".$emailSearchForString)->getBody();
+                $messages = json_decode($messages, true);
+            } while ($counter < 60 && $messages == Null);
+        }
         return array_shift($messages);
```

Comment:
The method `searchForMessage()` could be added in addition with the `fetchLastMessage` method, no need to get rid of it.

Reply:
Also:
- the `else` statement could be avoided
- the InboxID parameter should be optional, since it will default to the one in the config
- emails are usually received by mailtrap pretty fast, not sure doing 60 attempts is necessary
- the use of `sleep()` and `echo` is not recommended, nor necessary.